stdlib/socket/src/main/java/org/ballerinalang/stdlib/socket/tcp/SocketUtils.java

```diff
@@ -182,4 +185,17 @@ public static void shutdownExecutor(ExecutorService executorService) {
             executorService.shutdownNow();
         }
     }
+
+    /**
+     * This will return socket read timeout config by the user or default. Default value will be 5 minutes.
+     *
+     * @return Socket read timeout value in milliseconds
+     */
+    public static long getReadTimeout() {
+        long timeout = DEFAULT_READ_TIMEOUT_VALUE;
+        if (ConfigRegistry.getInstance().contains(CONFIG_READ_TIMEOUT)) {
```

Comment:
IMO reading config value as a runtime param should have the highest priority. WDYT?

Reply:
Normally closest config to the code should take the precedent. Also, this value affects globally.